next/src/field/schema.ts

```diff
@@ -1,18 +1,58 @@
-import type { JsfSchema } from '../types'
+import type { JsfObjectSchema, JsfSchema, NonBooleanJsfSchema } from '../types'
 import type { Field } from './type'
-import { buildFieldsObject } from './object'
+import { buildFieldObject } from './object'
 
-export function buildFieldsSchema(params: {
-  schema: JsfSchema
-}): Field[] {
-  const { schema } = params
+/**
+ * Get the input type for a field
+ * @param schema - The schema of the field
+ * @returns The input type for the field, based schema type. Default to 'text'
+ */
+function getInputType(schema: NonBooleanJsfSchema): string {
+  if (schema.type === 'string') {
+    return 'text'
+  }
+
+  return 'text'
+}
 
-  if (typeof schema === 'boolean')
-    return []
+/**
+ * Build a field from any schema
+ * @param schema - The schema of the field
+ * @param name - The name of the field, used if the schema has no title
+ * @param required - Whether the field is required
+ * @returns The field
+ */
+export function buildFieldSchema(
+  schema: JsfSchema,
+  name: string,
+  required: boolean = false,
+): Field | null {
+  if (typeof schema === 'boolean') {
+    return null
+  }
 
   if (schema.type === 'object') {
-    return buildFieldsObject({ schema })
+    const objectSchema: JsfObjectSchema = { ...schema, type: 'object' }
+    return buildFieldObject(objectSchema, name, required)
+  }
+
+  const inputType = getInputType(schema)
+
+  const field: Field = {
+    ...schema['x-jsf-presentation'],
+    inputType,
+    type: inputType,
+    name: schema.title || name,
+    required,
```

Comment:
todo: We need to spread the reaming `...schema` down. This [internal ticket](https://linear.app/remote/issue/RMT-921) explains why

Reply:
Do we have a list of keys that should not be spread? Is it just `['title', 'oneOf']`?

Reply:
In my opinion, we should spread everything, even `title` and `oneOf`. Minimize as few opinions as possible.

Reply:
P.S. I know, having field.title and field.label doesn't sound right, so let's leave title as the only one out for now. in the future i'd see this being customizable.

```js
createHeadlessForm(jsonSchema, {
  mapKeys: { title: 'label' }
}
```

So the Field would be indeed `label` but by default it should be `title` (without any opinionated renames)

Reply:
Sounds good, I've added it now as the first thing assigned to a field so that the spread `x-jsf-presentation` and the explicitly mapped keys like `inputType`, `name`, etc. take precedence.

``` typescript
  const { title: _title, ...spreadSchema } = schema

  const field: Field = {
    ...spreadSchema,
    ...schema['x-jsf-presentation'],
    inputType,
    type: inputType,
    jsonType: getJsonType(schema),
    name,
    required,
  }
```

Reply:
It's great, thanks!